src/Microsoft.AspNet.Owin/OwinFeatureCollection.cs

```diff
@@ -427,6 +428,20 @@ public bool IsReadOnly
             get { return true; }
         }
 
+        Guid IRequestIdentifierFeature.TraceIdentifier
+        {
+            get
+            {
+                var requestId = Prop<string>(OwinConstants.RequestId);
+                if (requestId != null)
+                {
+                    return new Guid(requestId);
```

Comment:
OWIN on the top of any of IIS/Systemweb/WebListener will not be a problem. But this can affect any other servers. The alternate that I can think of is change all 'Guid` to `string` and continue to produce `GUID` in our servers so that E2E tracing is possible (and just do a `guid.ToString()`)?